tests/bcl-test/templates/common/TestRunner.xUnit/XUnitTestRunner.cs

```diff
@@ -916,8 +936,10 @@ XElement Run (Assembly assembly, string assemblyPath)
 					executionOptions.SetSynchronousMessageReporting (true);
 
 					try {
+						// set the wait for event cb first, then execute the tests
+						var resultTask = WaitForEvent (resultsSink.Finished, TimeSpan.FromDays (10)).ConfigureAwait (false);
 						frontController.RunTests (testCases, resultsSink, executionOptions);
-						resultsSink.Finished.WaitOne ();
+						await resultTask;
```

Comment:
Do we need to continue on the main thread here?

```suggestion
						await resultTask.ConfigureAwait (false);
```

Reply:
You are 100% right, good catch (I was working late).